state/unit.go

```diff
@@ -603,6 +606,20 @@ func (op *DestroyUnitOperation) eraseHistory() error {
 	return nil
 }
 
+func (op *DestroyUnitOperation) deleteSecrets() error {
+	store := NewSecrets(op.unit.st)
+	ownedURIs, err := op.unit.st.referencedSecrets(op.unit.Tag(), "owner-tag")
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, uri := range ownedURIs {
+		if _, err := store.DeleteSecret(uri, nil); err != nil {
```

Comment:
It would be great to have a bulk deletion here.

Reply:
i added a helper so that internal state methods can do bulk deletion in a single transaction